src/main/kotlin/app/revanced/patches/reddit/testing/patch/TotalAdPostPatch.kt

```diff
@@ -0,0 +1,179 @@
+package app.revanced.patches.reddit.testing.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.extensions.removeInstruction
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprintResult
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod.Companion.toMutable
+import app.revanced.patches.reddit.testing.annotations.AdPostCompatibility
+import app.revanced.patches.reddit.testing.fingerprints.AdPostFingerprint
+import app.revanced.patches.reddit.testing.fingerprints.NewAdPostFingerprint
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.builder.MutableMethodImplementation
+import org.jf.dexlib2.dexbacked.reference.DexBackedFieldReference
+import org.jf.dexlib2.iface.instruction.formats.Instruction22c
+import org.jf.dexlib2.iface.instruction.formats.Instruction35c
+import org.jf.dexlib2.immutable.ImmutableMethod
+import org.jf.dexlib2.immutable.ImmutableMethodParameter
+
+@Patch
+@Name("hide-promoted")
+@Description("Removes promoted posts in all feeds.")
+@AdPostCompatibility
+@Version("0.0.1")
+class TotalAdPostPatch : BytecodePatch(
```

Comment:
This patch should be merged with the existing ads patch. Both serve the same purpose.

Reply:
Did not want to overwrite an alternative way to patch if this gets outdated :pensive:

Reply:
Ideally, patches are made with being long-lasting in mind. This patch currently uses fingerprints, which shouldn't break easily when Reddit updates. The implementations of the patch can be allocated to the integrations which would help with future stability.